My new branch (#37)
mojaloop/reference-architecture-doc: docs/refarch/boundedContexts/accountLookupAndDiscovery/index.md

# Account Lookup And Discovery BC

> The Accounts Lookup and Discovery BC is responsible for locating and associating participants and parties with party or participant triggered transactions.

Author: @elnyry-sam-k I am waiting for a response on the Oracle query from either @mdebarros or @pedrosousabarreto for this suggestion... thanks 😄

#### Description

> Where a Participant DFSP indicates Parties/Participants require association with a particular transaction, this UC is used by the Switch to validate the request and associate the Parties/Participants in accordance with the receipted request/instruction.

Reviewer: @elnyry-sam-k please review and provide your input here.

> Where a Participant DFSP indicates Parties/Participants require association with a particular transaction, this UC is used by the Switch to validate the request and associate the Parties/Participants in accordance with the receipted request/instruction.

Reviewer 2: Where a Participant DFSP is able to associate the provided Party identity(s). The checks and validations of the KYC (Know You Customer) details are not covered here and are left to processes outside of the Mojaloop API calls and should be covered by the Scheme, to ensure these association (or disassociation) requests are valid.

> Where a Participant DFSP indicates Parties/Participants require association with a particular transaction, this UC is used by the Switch to validate the request and associate the Parties/Participants in accordance with the receipted request/instruction.

Author: @elnyry-sam-k  and @mdebarros, I have put together a potential candidate for inclusion in the document using my original contrib. along with Sam's.  It reads as follows -

"_Where a Participant DFSP indicates Parties/Participants require association with a particular transaction, this UC is used by the Switch to validate the request and, where a Participant DFSP is able to associate the provided Party identity(s), associate the Parties/Participants in accordance with the receipted request/instruction.

Note: Checks and validations of the KYC (Know You Customer) details are not covered here and are left to processes outside of the Mojaloop API calls and should be covered by the Scheme, to ensure that association (or disassociation) requests are valid._"

Let me know what you think.  Thanks.

> Where a Participant DFSP indicates Parties/Participants require association with a particular transaction, this UC is used by the Switch to validate the request and associate the Parties/Participants in accordance with the receipted request/instruction.

Reviewer 2: looks good..

> Where a Participant DFSP indicates Parties/Participants require association with a particular transaction, this UC is used by the Switch to validate the request and associate the Parties/Participants in accordance with the receipted request/instruction.

Author: @elnyry-sam-k / @mdebarros This conversation can be resolved - I have updated my local branch and will push the change later today.  Thanks 😄

#### Description

> Where a Participant DFSP indicates Parties/Participants require disassociation with a particular transaction, this UC is used by the Switch to validate the request and disassociate the Parties/Participants in accordance with the receipted request/instruction.

Reviewer: @elnyry-sam-k please review and provide your input here.

> Where a Participant DFSP indicates Parties/Participants require disassociation with a particular transaction, this UC is used by the Switch to validate the request and disassociate the Parties/Participants in accordance with the receipted request/instruction.

Reviewer 2: looks good Michael

#### Description

> Where a participant DFSP requests Participant details, this UC is used by the switch to validate the request and provide the requested relevant participant data to the requesting DFSP.

Reviewer: ```suggestion
Where a Participant DFSP requests Participant association information based on a Parties identifier.
```

Reviewer: @elnyry-sam-k please review and provide your input here.

Author: @elnyry-sam-k & @mdebarros please let me know if you're good with this paragraph construct:

"Where a Participant DFSP requests Participant association information based on a Party identifier, this UC is used by the switch to validate the request and provide the requested participant data to the requesting DFSP."

Thanks 😄

Reviewer 2: this is good. thank you :-)

#### Description

> Where a participant DFSP requests Party details, this UC is used by the switch to validate the request and provide the requested relevant Party data to the requesting DFSP.

Reviewer: ```suggestion
Where a participant DFSP requests Party information based on a Parties identifier.
```

Reviewer: @elnyry-sam-k please review and provide your input here.

Reviewer 2: looks good Michael, thank you..

Reviewer 2: no problem Michael, have a great weekend; it's no problem, can continue till we resolve the open issues :-) ..
